fenix/app/src/main/java/org/mozilla/fenix/onboarding/view/JunoOnboardingScreen.kt

```diff
@@ -0,0 +1,181 @@
+import com.google.accompanist.pager.rememberPagerState
+import kotlinx.coroutines.launch
+import mozilla.components.lib.state.ext.observeAsComposableState
+import org.mozilla.fenix.compose.PagerIndicator
+import org.mozilla.fenix.compose.annotation.LightDarkPreview
+import org.mozilla.fenix.ext.components
+import org.mozilla.fenix.theme.FirefoxTheme
+
+private val OnboardingPageTypeList = listOf(
+    JunoOnboardingPageType.DEFAULT_BROWSER,
+    JunoOnboardingPageType.NOTIFICATION_PERMISSION,
+    JunoOnboardingPageType.SYNC_SIGN_IN,
+)
+
+/**
+ * A screen for displaying juno onboarding.
+ *
+ * @param onMakeFirefoxDefaultClick Invoked when the positive button on default browser page is
+ * clicked.
+ * @param onPrivacyPolicyClick Invoked when the privacy policy link text is clicked.
+ * @param onSignInButtonClick Invoked when the positive button on the sign in page is clicked.
+ * @param grantNotificationPermissionButtonClick Invoked when the positive button on notification
+ * page is clicked.
+ * @param onFinish Invoked when the onboarding is completed.
+ */
+@Composable
+fun JunoOnboardingScreen(
+    onMakeFirefoxDefaultClick: () -> Unit,
+    onPrivacyPolicyClick: (String) -> Unit,
+    onSignInButtonClick: () -> Unit,
+    grantNotificationPermissionButtonClick: () -> Unit,
+    onFinish: () -> Unit,
+) {
+    val context = LocalContext.current
+    val coroutineScope = rememberCoroutineScope()
+    val pagerState = rememberPagerState()
+    val isSignedIn: State<Boolean?> = context.components.backgroundServices.syncStore
+        .observeAsComposableState { it.account != null }
+
+    BackHandler(enabled = pagerState.currentPage > 0) {
+        coroutineScope.launch {
+            pagerState.animateScrollToPage(pagerState.currentPage - 1)
+        }
+    }
+
+    val scrollToNextPageOrDismiss: () -> Unit = {
+        if (pagerState.currentPage == pagerState.pageCount - 1) {
+            onFinish()
+        } else {
+            coroutineScope.launch {
+                pagerState.animateScrollToPage(pagerState.currentPage + 1)
+            }
+        }
+    }
+
+    LaunchedEffect(isSignedIn.value) {
+        if (isSignedIn.value == true) {
+            scrollToNextPageOrDismiss()
+        }
+    }
+
+    JunoOnboardingContent(
+        pagerState = pagerState,
+        scrollToNextPageOrDismiss = scrollToNextPageOrDismiss,
+        onMakeFirefoxDefaultClick = onMakeFirefoxDefaultClick,
+        onPrivacyPolicyClick = onPrivacyPolicyClick,
+        onSignInButtonClick = onSignInButtonClick,
+        grantNotificationPermissionButtonClick = grantNotificationPermissionButtonClick,
+    )
+}
+
+@Composable
+@Suppress("LongParameterList")
+private fun JunoOnboardingContent(
+    pagerState: PagerState,
+    scrollToNextPageOrDismiss: () -> Unit,
+    onMakeFirefoxDefaultClick: () -> Unit,
+    onPrivacyPolicyClick: (String) -> Unit,
+    onSignInButtonClick: () -> Unit,
+    grantNotificationPermissionButtonClick: () -> Unit,
```

Comment:
```suggestion
    onGrantNotificationPermissionButtonClick: () -> Unit,
```
We should ideally just follow our callback "on<X>" convention.